src/GifWin/ViewModels/GifWitConversionProgressViewModel.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Threading;
+using GalaSoft.MvvmLight;
+using GifWin.Data;
+
+namespace GifWin.ViewModels
+{
+    class GifWitConversionProgressViewModel : ViewModelBase, IProgress<int>
+    {
+        GifWitLibrary libraryToConvert;
+        GifWitConversionProgressWindow window;
+
+        public GifWitConversionProgressViewModel (GifWitLibrary libraryToConvert, GifWitConversionProgressWindow window)
+        {
+            this.libraryToConvert = libraryToConvert;
+            this.window = window;
+        }
+
+        int converted;
+        public int Converted
+        {
+            get { return converted; }
+            set { converted = value; RaisePropertyChanged (); }
+        }
+
+        internal void ConvertLibrary ()
+        {
+            Task.Run (async () => {
+                using (var db = new GifWinDatabaseHelper ()) {
+                    int converted;
+
+                    try {
+                        converted = await db.ConvertGifWitLibraryAsync (libraryToConvert, this);
+
+                        window.Dispatcher.Invoke (() => {
```

Comment:
Have `ConvertLibrary()` just convert the library and split deleting into its own method. The UI should present the UI, the VM should just offer the data and API to do it. The library to delete is provided on creation, so there really shouldn't be any issue disconnecting the convert action from the delete action.

Further make `ConvertLibrary` return a task for the conversion (and the deletion). The UI should be the thing that awaits on it and then does something with it.